viewing_party/party.py

```diff
@@ -1,23 +1,266 @@
+    for friend in friends_list: 
+        movie_list_each_friend = friend["watched"]
+        get_value_list_with_same_key_in_dict_list(movie_list_each_friend, "title", titles_friends)
         
+    # compare 
+    movie_set_user = set(titles_user)
+    movie_set_friends = set(titles_friends)
+    unique_movie_title_friends = movie_set_friends - movie_set_user
+
+    # list of unique movies (in dict)
+    unique_movie_friends = []
+    for friend in friends_list: 
+        movie_list_each_friend = friend["watched"]
+        for movie in movie_list_each_friend:
+            title = movie["title"]
+            # avoid duplicate!
+            if title in unique_movie_title_friends and not movie in unique_movie_friends:
+                unique_movie_friends.append(movie)
+        
+    return unique_movie_friends
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    '''
+    (1) service list of user subscribes
+    (2) friends' unique movie list: call get_friends_unique_watched
+    (3-1) check if services of (2) in (1)
+    (3-2) recommended list: list of friends' unique movies on a services which user subscribes
+    '''
+
+    service_list_user = user_data["subscriptions"]
+    unique_movie_friends = get_friends_unique_watched(user_data) # in list
+    recommended_list = []
+
+    # friend in dict
+    for movie in unique_movie_friends: 
+        service_friends = movie["host"]
+        if service_friends in service_list_user:
+            recommended_list.append(movie)
+    
+    return recommended_list
+    
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    '''
+    (1) user's most frequently watched genre: call get_most_watched_genre(user_data)
+        ! assume no multiple genre
+    (2) friends' unique movie list: call get_friends_unique_watched
+    (3) compare genre of the movie to user's favorite
+    '''
+
+    most_genre = get_most_watched_genre(user_data) # in str
+    unique_movie_friends = get_friends_unique_watched(user_data) # in dict
+    recommended_list = []
+
+    for movie in unique_movie_friends: 
+        genre_friends = movie["genre"]
+        if genre_friends == most_genre:
+            recommended_list.append(movie)
+    
+    return recommended_list
+
+def get_rec_from_favorites(user_data):
+    '''
+    (1) favorite movie list (elements are in dict)
+    (2) user's unique movie list: call get_unique_watched
+    (3) favorite movies in user's unique list (in a list of dict)
+    '''
+
+    favorite_list = user_data["favorites"]
+    titles_user_favorite = []
+    recommended_list = []
+    unique_movie_user = get_unique_watched(user_data)
+    get_value_list_with_same_key_in_dict_list(favorite_list, "title", titles_user_favorite)

```

Comment:
This seems like a lot of work! For the if statement on line 263, it would be possible to do `if movie in user_data["favorites"]`. This will take away the need to just pull out the titles, you can just check if the specific movie dictionary is in the list.